lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -778,21 +780,52 @@ impl TryFrom<BitmapEvent> for CGOBitmap {
             data_cap: 0,
         };
 
-        // e.decompress consumes e, so we need to call it separately, after populating the fields
-        // above.
-        let mut data = e.decompress()?;
-        res.data_ptr = data.as_mut_ptr();
-        res.data_len = data.len();
-        res.data_cap = data.capacity();
+        let w: u16 = e.width;
+        let h: u16 = e.height;
+
+        let mut encoded = Vec::with_capacity(8192);
```

Comment:
I also notice that we're reallocating here for each PNG. It looks like a place where we might squeeze out more performance by reusing a buffer attached to `Client`. It will require some care to avoid a memory leak, given that we're using this same memory on the Go side.

Reply:
I'll try to check how much time it takes to do it in this way and whether it is worth adding the complexity of the shared buffer. Thanks for suggestion!